docs/contributing.rst

```diff
@@ -69,6 +69,45 @@ found in the `virtualenv docs`_.
 
 .. _`virtualenv docs`: https://virtualenv.pypa.io
 
+Running certbot against a locally deployed Pebble CA
```

Comment:
I'd move this to be a new section under "Testing" after "Integration testing with the Pebble CA". My thinking here is that while this is undoubtedly useful, I think this is the least important part of testing since it's just manual testing so it doesn't need to come so early in the doc.

In this new context, it changes the titles a bit. "Integration testing with the Pebble CA" could be something like "Running automated integration tests" and this section could be called "Running manual integration tests".

What do you think?

Reply:
What you propose was my first move. Then thinking of it again, I asked myself this question: as a contributor, willing to add a functionality, what would be the first thing I would do after writing the code?

I think it is: I would execute certbot to check my functionality is working.

In fact, we could consider that calling certbot to check the functionality is not part of the Quality Assurance process, it is part of the development workflow, and what I add here is a tool to help you doing this step. 

In that spirit, running the unit tests and the integration tests is part of the QA, and will maybe be executed by the contributor, or inevitably by the CI that will yell at you back when the PR is opened ^^

So in this case, the tool added here should be part of the dev section, right where I placed it, and unit/integration tests are in the testing part.

Does it make sense to you?

Reply:
I understand where you're coming from, but I still would like to see this change for a couple reasons.

The first reason is that it breaks the flow of the document, at least in some cases. For example, when we run Certbot hackathons, we usually tell people to go to this doc. In master, the changes are described in this order:

1. Set up the development environment.
2. Find an issue to work on.
3. Test it.

I think this is the correct workflow as there's not much to test against Pebble until you have something to work on.

The second reason is I suspect most core Certbot devs won't do this and therefore we shouldn't make these instructions so prominent. I suspect they'll rely more heavily on automated testing or manually testing against staging.

For those reasons, I'll suggest again you make the changes I described above, but as a compromise, I think you could add links to this section at the end of "Running a local copy of the client" and/or you could add a brief intro to the top of the "Testing" section with links to each section.

Reply:
Indeed, I can confirm your suspicion about core developers, at least on my side ^^

I agree with your compromise. Then I will put an intro on top of the Testing section. I think in this case it is the clearest thing to do.

Reply:
Done.